firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java

```diff
@@ -145,10 +121,33 @@ boolean check(PerfMetric metric) {
     } else if (metric.hasTraceMetric()) {
       return traceLimiter.check(metric);
     } else {
+      // Should not reach here
       return false;
     }
   }
 
+  /**
+   * Check if the {@link PerfMetric} should be sampled. A {@link PerfMetric} is considered sampled
+   * if the device isn't allowed to send the event type and it is not part of a verbose session.
+   *
+   * @param metric {@link PerfMetric} object.
+   * @return true if allowed, false if not allowed.
+   */
+  boolean checkDeviceSampling(PerfMetric metric) {
```

Comment:
Nit: Can you rename it as `isDeviceSampled`?

Reply:
Done